tests/compile-test.rs

```diff
@@ -155,38 +155,37 @@ fn default_config() -> compiletest::Config {
     config
 }
 
-fn run_ui(cfg: &mut compiletest::Config) {
+fn run_ui(mut cfg: compiletest::Config) {
     cfg.mode = TestMode::Ui;
     cfg.src_base = Path::new("tests").join("ui");
+    cfg.build_base.push("ui");
     // use tests/clippy.toml
     let _g = VarGuard::set("CARGO_MANIFEST_DIR", std::fs::canonicalize("tests").unwrap());
-    compiletest::run_tests(cfg);
+    compiletest::run_tests(&cfg);
 }
 
-fn run_ui_test(cfg: &mut compiletest::Config) {
+fn run_ui_test(mut cfg: compiletest::Config) {
     cfg.mode = TestMode::Ui;
     cfg.src_base = Path::new("tests").join("ui_test");
+    cfg.build_base.push("ui_test");
     let _g = VarGuard::set("CARGO_MANIFEST_DIR", std::fs::canonicalize("tests").unwrap());
     let rustcflags = cfg.target_rustcflags.get_or_insert_with(Default::default);
-    let len = rustcflags.len();
     rustcflags.push_str(" --test");
-    compiletest::run_tests(cfg);
-    if let Some(ref mut flags) = &mut cfg.target_rustcflags {
-        flags.truncate(len);
```

Comment:
Any idea what this was for?

Reply:
Oh, it's because the config isn't reused right?

Reply:
Yeah, it was to undo the config changes